popup.js

```diff
@@ -57,57 +62,103 @@ function processLatestTransactions(latestTransactions = []) {
                     txInfo.innerHTML += "BlockchainDNS Registration";
-                    
+
                     txType.innerHTML = "REG";
                     txType.style.color = "rgb(30, 118, 210)";
 
                     continue;
-                } else if (latestTransactions[i].amount === -0.005 && latestTransactions[i].fee === -0.005) {
+                } else if (currentTransactionAmount === -0.005 && latestTransactions[i].fee === -0.005) {
                     txInfo.innerHTML += "BlockchainDNS Update";
-                    
+
                     txType.innerHTML = "UPD";
                     txType.style.color = "rgb(30, 118, 210)";
 
                     continue;
                 }
             }
 
-            if (latestTransactions[i].category == "generate" || latestTransactions[i].category == "receive") {
-                txType.innerHTML = "IN";
-                txType.style.color = "rgb(50, 205, 50)";
-            } else if (latestTransactions[i].category === "send") {
-                txType.innerHTML = "OUT";
-                txType.style.color = "rgb(255, 0, 0)";
-            } else if (latestTransactions[i].category == "immature") {
-                txType.innerHTML = "IMM";
-                txType.style.color = "rgb(192, 192, 192)";
-            } else if (latestTransactions[i].category == "orphan") {
-                txType.innerHTML = "OPH";
-                txType.style.color = "rgb(192, 192, 192)";
+            switch (transactionCategory) {
+                case "generate": case "receive":
+                    txType.innerHTML = "&#8592;";
+                    txType.style.color = "rgb(50, 205, 50)";
+                    break;
+                case "send":
+                    if (isToSelf(i, currentTransactionAddress)) {
+                        txType.innerHTML = "&#8651;";
+                        txType.style.color = "rgb(138,43,226)";
+                        i--;
+                    } else {
+                        txType.innerHTML = "&#8594;";
+                        txType.style.color = "rgb(255, 0, 0)";
+                    }
+                    break;
+                case "immature":
+                    txType.innerHTML = "IMM";
+                    txType.style.color = "rgb(192, 192, 192)";
+                    break;
+                case "orphan":
+                    txType.innerHTML = "OPH";
+                    txType.style.color = "rgb(192, 192, 192)";
+                    break
+                default:
+                    txType.innerHTML = "ERR";
+                    txType.style.color = "rgb(255, 0, 0)";
             }
         }
     } else {
         document.getElementById("tx-info-0").innerHTML = "&emsp;&emsp;&emsp;&emsp;No transaction history.";
     }
+
+    function isGenerateOrReceive(transactionIndex) {
+        const currentTransactionCategory = latestTransactions[transactionIndex].category;

```

Comment:
do we really need this const and even the function? aren't self transfers pairs of send-receive? send-generate souldn't really apply